### PR TITLE
hdf5: add option enable_unsupported

### PR DESCRIPTION
**Pull request by joakimono**
Specify library name and version:  **hdf5/all**

This PR exposes a new option **enable_unsupported**, which translates to ALLOW_UNSUPPORTED in CMakeLists. The default option retains the default behavior, but now allows package consumers to build previously disallowed option configurations.
The commit also fixes some minor issues with the recipe.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/hdf5/all/conanfile.py [hunk 1]
```diff
@@ -278,11 +283,11 @@ def _config_libname(lib):
             self.cpp_info.components[component_name].requires = requirements
             self.cpp_info.components[component_name].includedirs.append(os.path.join("include", "hdf5"))
 
-            # TODO: to remove in conan v2 once cmake_find_package_* generators removed
-            self.cpp_info.components[component_name].names["cmake_find_package"] = component
-            self.cpp_info.components[component_name].names["cmake_find_package_multi"] = component
-            self.cpp_info.components[component_name].build_modules["cmake_find_package"] = [self._module_targets_file_rel_path, self._module_variables_file_rel_path]
-            self.cpp_info.components[component_name].build_modules["cmake_find_package_multi"] = [self._module_targets_file_rel_path, self._module_variables_file_rel_path]
+            if conan_version.major == "1":
```
Comment (prince-chrismc) [inline recipes/hdf5/all/conanfile.py:286]: Please revert this is not required :)

File: recipes/hdf5/all/conanfile.py [hunk 2]
```diff
@@ -172,6 +176,7 @@ def generate(self):
         if Version(self.version) >= "1.10.0":
             tc.variables["HDF5_BUILD_JAVA"] = False
         # Honor BUILD_SHARED_LIBS from conan_toolchain (see https://github.com/conan-io/conan/issues/11840)
+        tc.variables["ALLOW_UNSUPPORTED"] = self.options.enable_unsupported
```
Comment (prince-chrismc) [inline recipes/hdf5/all/conanfile.py:179]: This line breaks up the comment above with the code its referring to

I would suggest deleting both and bump the requirement for the Conan version at the top of the recipe to 1.54.0

File: recipes/hdf5/all/conanfile.py [hunk 3]
```diff
@@ -59,13 +62,13 @@ def validate(self):
         if not can_run(self):
             # While building it runs some executables like H5detect
             raise ConanInvalidConfiguration("Current recipe doesn't support cross-building (yet)")
-        if self.info.options.parallel:
-            if self.info.options.enable_cxx:
+        if self.options.parallel and not self.options.enable_unsupported:
+            if self.options.enable_cxx:
                 raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
```
Comment (prince-chrismc) [inline recipes/hdf5/all/conanfile.py:67]: Should the comment be updated to include the new option?

File: recipes/hdf5/all/conanfile.py (hunk 1 above)
Comment (joakimono) [reply recipes/hdf5/all/conanfile.py]: Reverted

File: recipes/hdf5/all/conanfile.py (hunk 2 above)
Comment (joakimono) [reply recipes/hdf5/all/conanfile.py:175]: Fixed

File: recipes/hdf5/all/conanfile.py (hunk 3 above)
Comment (joakimono) [reply recipes/hdf5/all/conanfile.py]: Yes, I think it should. Done.

**Comment by prince-chrismc** [comment]
There's merge conflcits FYI

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request closed without merging**